tensorflow/python/layers/utils.py

```diff
@@ -207,14 +207,21 @@ def constant_value(pred):
 
     Arguments:
       pred: A scalar, either a Python bool or a TensorFlow boolean variable
-        or tensor.
+        or tensor, or the Python integer 1 or 0.
 
     Returns:
       True or False if `pred` has a constant boolean value, None otherwise.
 
     Raises:
-      TypeError: If `pred` is not a Variable, Tensor or bool.
+      TypeError: If `pred` is not a Variable, Tensor or bool, or Python
+        interger 1 or 0.
     """
+  # Allow integer booleans.
+  if pred == 0:
```

Comment:
Should this be after the Variable check? There's a danger you'll get back a boolean Tensor if pred is a Variable?

Reply:
Or alternatively, should we check for type(pred) is int before we do this?

Reply:
This is the original code during the review of the PR. I can do that as well

Reply:
Yeah I already did that. This PR took much more time than I thought before